service/matching/workers/registry_impl.go

```diff
@@ -116,6 +117,23 @@ func (b *bucket) filterWorkers(
 	return out
 }
 
+func (b *bucket) getWorkerHeartbeat(nsID namespace.ID, workerInstanceKey string) (*workerpb.WorkerHeartbeat, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	mp, ok := b.namespaces[nsID]
+	if !ok {
+		return nil, serviceerror.NewNotFoundf("namespace not found: %s", nsID.String())
```

Comment:
we actually have a more specific serviceerror.NewNamespaceNotFound(ns). should we use that instead? we have only an id instead of a name but maybe it's okay?